### PR TITLE
Add support for LT7841

### PR DESCRIPTION
**Pull request by MCabuena**
## Pull Request Description

Please replace this with a detailed description and motivation of the changes. 
You can tick the checkboxes below with an 'x' between square brackets or just check them after publishing the PR. 
If this PR contains a breaking change, list dependent PRs and try to push all related PRs at the same time.

## PR Type
- [ ] Bug fix (change that fixes an issue)
- [x] New feature (change that adds new functionality)
- [ ] Breaking change (has dependencies in other repos or will cause CI to fail)

## PR Checklist
- [x] I have followed the [Coding style guidelines](http://analogdevicesinc.github.io/no-OS/drivers_guide.html#coding-style)
- [x] I have performed a self-review of the changes
- [x] I have commented my code, at least hard-to-understand parts
- [x] I have build all projects affected by the changes in this PR
- [x] I have tested in hardware affected projects, at the relevant boards
- [x] I have signed off all commits from this PR
- [x] I have updated the documentation (wiki pages, ReadMe etc), if applies

### CODE REVIEW

**Comment by buha** [comment]
there are lots of CI issues, please fix those before a review

**Comment by MCabuena** [comment]
V2:

1. used astyle.exe with settings set from https://github.com/analogdevicesinc/no-OS/blob/main/ci/astyle_config
2. removed i2c.h from ltc7841.h file
3. corrected .rst issues from README.rst of driver folder

**Comment by kister-jimenez** [comment]
@MCabuena Please rebase from the latest main so we can run the pipeline.

**Comment by MCabuena** [comment]
V3:

1.  rebased commit to updated commit of main branch

**Comment by MCabuena** [comment]
V4:

1. fix README rst warning for project file
2. fix astyle tab to be equal to 8 spaces

**Comment by MCabuena** [comment]
V5:

1. corrected comment for iio_ltc7841.c using desc instead of iio_desc
2. corrected comment for ltc7841.c of using device instead of desc

File: projects/ltc4296/src/common/common_data.c
Comment (kister-jimenez) [inline projects/ltc4296/src/common/common_data.c:1]: This is a mistake and should not be included in this commit.

File: doc/sphinx/source/drivers_doc.rst [hunk 1]
```diff
@@ -104,6 +104,7 @@ POWER MANAGEMENT
    drivers/lt8722
    drivers/ltp8800
    drivers/max42500
+   drivers/ltc7841
```
Comment (kister-jimenez) [inline doc/sphinx/source/drivers_doc.rst:107]: sort alphabetically

File: doc/sphinx/source/projects_doc.rst [hunk 2]
```diff
@@ -77,6 +77,7 @@ POWER MANAGEMENT
    projects/lt8722
    projects/ltp8800
    projects/max42500
+   projects/ltc7841
```
Comment (kister-jimenez) [inline doc/sphinx/source/projects_doc.rst:80]: Sort alphabetically

File: drivers/power/ltc7841/ltc7841.c [hunk 3]
```diff
@@ -0,0 +1,480 @@
...
+	if (vout_level <= MFR_MAX_VALUE) {
+		// okay to proceed
+		ret = ltc7841_reg_write(desc, LTC7841_MFR_VOUT_COMMAND, vout_level);
+		if (ret) {
+			return ret;
+		} else {
+			ret = -1;
+		}
+	}
+	return ret;
```
Comment (kister-jimenez) [inline drivers/power/ltc7841/ltc7841.c:404]: CI checks fail here because outer if has no else therefore, ret has a chance of not getting initialized.

File: doc/sphinx/source/drivers_doc.rst [hunk 4]
```diff
@@ -98,10 +98,11 @@ POWER MANAGEMENT
    drivers/adp1050
    drivers/ltc2992
    drivers/ltc4296
+   drivers/ltc7841
    drivers/lt7170
    drivers/lt7182s
    drivers/lt8491
-   drivers/lt8722
+   drivers/lt8722
```
Comment (kister-jimenez) [inline doc/sphinx/source/drivers_doc.rst:105]: Avoid unrelated changes.

**Review by kister-jimenez — COMMENTED** [review]
You can build the documentations inside the doc/sphinx/source so you can check if you have errors before you push. You can check the CI scripts for the commands or you can also run scripts in the CI folder locally so you can check from your side if you installed the dependencies.

File: drivers/power/ltc7841/ltc7841.c (hunk 3 above)
Comment (MCabuena) [reply drivers/power/ltc7841/ltc7841.c:404]: I didn't write my code this way, formatting with astyle.exe caused this issue.

My original code was :
int ltc7841_change_vout_command(struct ltc7841_desc *desc, uint16_t vout_level)
{
	int ret;

	if (vout_level <= MFR_MAX_VALUE) {
		// okay to proceed
		ret = ltc7841_reg_write(desc, LTC7841_MFR_VOUT_COMMAND, vout_level);
		if (ret) 
		{
			return ret;
		} 
	}
	else
	{
		ret = -1;
	}
	return ret;
}

astyle formatting made it so that the outer else turned into an inner else for some reason

Comment (MCabuena) [reply drivers/power/ltc7841/ltc7841.c:404]: I'm changing it now and will double check after formatting with astyle to be sure

Comment (kister-jimenez) [reply drivers/power/ltc7841/ltc7841.c:404]: Also, don't put {} if there is only a single statement.

Please use the errors defined in errno.h.

Another thing, macros should be prepended with LTC7841 i.e. MFR_MAX_VALUE -> LTC7841_MFR_MAX_VALUE
